core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -65,23 +73,63 @@
 @Service
 public class SpecService {
 
+  private final EntityRepository entityRepository;
   private final FeatureSetRepository featureSetRepository;
   private final ProjectRepository projectRepository;
   private final StoreRepository storeRepository;
   private final Source defaultSource;
 
   @Autowired
   public SpecService(
+      EntityRepository entityRepository,
       FeatureSetRepository featureSetRepository,
       StoreRepository storeRepository,
       ProjectRepository projectRepository,
       Source defaultSource) {
+    this.entityRepository = entityRepository;
     this.featureSetRepository = featureSetRepository;
     this.storeRepository = storeRepository;
     this.projectRepository = projectRepository;
     this.defaultSource = defaultSource;
   }
 
+  /**
+   * Get an entity matching the entity name and set project. The entity name and project are
+   * required. If the project is omitted, the default would be used.
+   *
+   * @param request GetEntityRequest Request
+   * @return Returns a GetEntityResponse containing an entity
+   */
+  public GetEntityResponse getEntity(GetEntityRequest request) {
+    String projectName = request.getProject();
+    String entityName = request.getName();
+
+    if (entityName.isEmpty()) {
+      throw new IllegalArgumentException("No entity name provided");
+    }
+    // Autofill default project if project is not specified
+    if (projectName.isEmpty()) {
+      projectName = Project.DEFAULT_NAME;
+    }
+
+    checkValidCharacters(projectName, "project");
```

Comment:
do we need to validate it here?
we assume that entity with invalid characters cannot be created

Reply:
Not necessary as we could just return nothing found. Just thought this would be clearer to the user, telling them that they're retrieving entity with invalid characters.